certbot/reverter.py

```diff
@@ -223,53 +225,33 @@ def _add_to_checkpoint_dir(self, cp_dir, save_files, save_notes):
             cp_dir, constants.CONFIG_DIRS_MODE, compat.os_geteuid(),
             self.config.strict_permissions)
 
-        op_fd, existing_filepaths = self._read_and_append(
-            os.path.join(cp_dir, "FILEPATHS"))
+        storage_path = os.path.join(cp_dir, "FILEPATHS")
+        existing_filepaths = _read_json_file(storage_path)
 
         idx = len(existing_filepaths)
 
-        for filename in save_files:
-            # No need to copy/index already existing files
-            # The oldest copy already exists in the directory...
-            if filename not in existing_filepaths:
-                # Tag files with index so multiple files can
-                # have the same filename
-                logger.debug("Creating backup of %s", filename)
-                try:
+        try:
```

Comment:
Why change the scope of this try/except block? In general, I think it's good practice to limit the scope to the code you expect to raise the error you're trying to handle.

Reply:
After live discussion, I see that the goal here is to write as much as possible in the storage file if something goes wrong.

I use now the original try except approach on each file, and write to the storage file immediately after having copied the file content in a safe location.